recipes/_configure_server.rb

```diff
@@ -41,13 +41,35 @@
 case node['splunk']['node_type']
 when :search_head, :server
   clusters = CernerSplunk.all_clusters(node).collect do |(cluster, bag)|
-    stanza = "clustermaster:#{cluster}"
     master_uri = bag['master_uri'] || ''
+    deployer_uri = bag['deployer_uri'] || ''
     settings = bag['settings'] || {}
     pass = settings['pass4SymmKey'] || ''
+    shc_replication_factor = settings['_cerner_splunk_shc_rep_factor'] || ''
+
+    if node['splunk']['config']['sh_cluster']
+
+      fail "Deployer URI is not specified for #{cluster}" if deployer_uri.empty?
+
+      replication_ports = bag['replication_ports'] || {}
+      replication_ports.each do |port, port_settings|
+        ssl = port_settings['_cerner_splunk_ssl'] == true
+        stanza = ssl ? "replication_port-ssl://#{port}" : "replication_port://#{port}"
+        server_stanzas[stanza] = port_settings.delete_if do |k, _|
```

Comment:
use reject instead of delete_if so you aren't modifying the underlying configuration hash. I fixed a bunch of these in my last PR (https://github.com/cerner/cerner_splunk/commit/654b2d5bbbcbad2e48d2e172637bf4c1d5cd17f0#diff-099855ec276c01c66dd6f3b1266ef1b0).